js/main.js

```diff
@@ -1,37 +1,74 @@
-var $newEntry = document.querySelector('form');
+var $entry = document.querySelector('form');
 var $previewPhoto = document.querySelector('img');
 var $photoURL = document.querySelector('#photo');
 $photoURL.addEventListener('input', photoURLBox);
 
+// preview photo in box
+
 function photoURLBox(event) {
-  $previewPhoto.setAttribute('src', $newEntry.elements.photo.value);
+  $previewPhoto.setAttribute('src', $entry.elements.photo.value);
 }
 
+// submit new entry and load into entry page
+
 function newEntry(event) {
-  var title = $newEntry.elements.title.value;
-  var photoURL = $newEntry.elements.photo.value;
-  var notes = $newEntry.elements.notes.value;
-  var newEntryObject = {
-    title,
-    photoURL,
-    notes,
-    entryId: data.nextEntryId
-  };
-
-  data.nextEntryId++;
-  data.entries.unshift(newEntryObject);
+  var title = $entry.elements.title.value;
+  var photoURL = $entry.elements.photo.value;
+  var notes = $entry.elements.notes.value;
+
+  var $header = document.querySelector('h1');
+
+  if ($header.textContent === 'New Entry') {
```

Comment:
I wouldn't base this on the text content of the DOM, you should instead base it on if `data.editing` currently has a value other than `null` or not.